### DISCUSSION TITLE
I defined a ruleset requiring for a conditional check to pass for a PR to be merged, but it checks all PRs and blocks irrelevant PRs to the check

### DISCUSSION THREAD

**Discussion by byungjoon72** [community/community#139540] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello team. I Defined a ruleset that requires a conditional check to pass for a PR to be merged.

<img width="769" alt="Screenshot 2024-09-23 at 2 09 13 PM" src="https://github.com/user-attachments/assets/c52f9812-6359-409d-aa9c-ec8607f78ef2">

The check is defined as a GitHub workflow, and that workflow is triggered only when a PR for certain files is published. 

<img width="448" alt="Screenshot 2024-09-23 at 2 07 38 PM" src="https://github.com/user-attachments/assets/3e9a8900-ebed-40e8-a1a3-c995db2be450">

But interestingly this ruleset is evaluated for all PRs (screenshot attached: "Java and Spark PR checker" in the pic) and stays in the "Expected" status forever. 

<img width="947" alt="Screenshot 2024-09-23 at 1 51 41 PM" src="https://github.com/user-attachments/assets/c905307a-1d67-4564-bead-d59de33e9df9">

Could you kindly help me know to address this issue?

**Comment by hankhsu1996** [comment, marked as answer]
This issue has been present in GitHub Actions for over two years. The official solution has changed several times, and the [current recommendation](https://docs.github.com/en/enterprise-cloud@latest/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks#handling-skipped-but-required-checks) is to avoid adding any past filter to the workflow. This means the job must be triggered unconditionally. Also, see [this discussion](https://github.com/orgs/community/discussions/13690).

A workaround to save compute resources is to use path filters in jobs or steps. If you detect that the change is unrelated to this CI, you can force the job to pass. Tools like [path-filter](https://github.com/dorny/paths-filter) or [changed-files](https://github.com/tj-actions/changed-files) can help implement this approach.

**Reply by byungjoon72** [reply]
Thank you @hankhsu1996. I think `path-filter` seems best for our purpose.
